File: ruby/test/fixtures/prompts/greet.rb
prompt description: "Generate a greeting",
       arguments: {
         name: "string",
         style: { type: "string", description: "Greeting style", optional: true }
       }

render do |args|
  style = args['style'] || 'friendly'
  [
    { "role" => "user", "content" => { "type" => "text", "text" => "Say hello to #{args['name']} in a #{style} way." } }
  ]
end

File: tests/conformance/test-prompts-php/greet.php
<?php
return [
    'description' => 'Greeting prompt',
    'arguments' => [
        'name' => 'string',
        'tone' => ['type' => 'string', 'optional' => true, 'description' => 'formal or casual'],
    ],
    'render' => function($args) {
        $name = $args['name'] ?? 'world';
        $tone = $args['tone'] ?? 'casual';
        return [
            ['role' => 'user', 'content' => ['type' => 'text', 'text' => "Greet {$name} in a {$tone} tone"]],
        ];
    },
];
